src/main/scala/io/qbeast/spark/table/IndexedTable.scala

```diff
@@ -84,203 +76,182 @@ trait IndexedTableFactory {
    * It is not guaranteed that the returned table physically
    * exists, use IndexedTable#exists attribute to verify it.
    *
-   * @param sqlContext the SQLContext to associate with the table
-   * @param path the table path
+   * @param tableId the table path
    * @return the table
    */
-  def getIndexedTable(sqlContext: SQLContext, path: String): IndexedTable
+  def getIndexedTable(tableId: QTableID): IndexedTable
 }
 
 /**
  * Implementation of IndexedTableFactory.
  *
  * @param keeper the keeper
- * @param oTreeAlgorithm the OTreeAlgorithm instance
  */
 final class IndexedTableFactoryImpl(
     private val keeper: Keeper,
-    private val oTreeAlgorithm: OTreeAlgorithm)
+    private val indexManager: IndexManager[DataFrame],
+    private val metadataManager: MetadataManager[StructType, FileAction],
+    private val dataWriter: DataWriter[DataFrame, StructType, FileAction],
+    private val revisionBuilder: RevisionFactory[StructType])
     extends IndexedTableFactory {
 
-  override def getIndexedTable(sqlContext: SQLContext, path: String): IndexedTable =
-    new IndexedTableImpl(sqlContext, path, keeper, oTreeAlgorithm)
+  override def getIndexedTable(tableID: QTableID): IndexedTable =
+    new IndexedTableImpl(
+      tableID,
+      keeper,
+      indexManager,
+      metadataManager,
+      dataWriter,
+      revisionBuilder)
 
 }
 
 /**
  * Implementation of IndexedTable.
  *
- * @param sqlContext the associated SQLContext
- * @param path the table path
+ * @param tableID the table identifier
  * @param keeper the keeper
- * @param oTreeAlgorithm the OTreeAlgorithm instance
  */
 private[table] class IndexedTableImpl(
```

Comment:
Update scaladocs

Reply:
> Everything looks good to me, but I think we should do a final improvement on scaladocs. These are "big" changes, or files containing several missing scaladocs (sorry in advancesee_no_evil):
> 
> * `io.qbeast.core.transform` classes from qbeast-core project.
> * class `QbeastDataSource` is also missing an argument in scaladocs, but I can't comment on it as it has not been modified with the refactoring.
> * scaladocs also missing on methods inside `private[table] class IndexedTableImpl` within `IndexedTable.scala`
> 
> The rest of the changes is small, you can see them below:

Big thank you Eric! Stupid question: do you know about an easy way of checking which scaladoc are missing in big projects?